src/Confluent.SchemaRegistry/Rest/Authentication/IHttpAuthenticator.cs

```diff
@@ -0,0 +1,34 @@
+﻿// Copyright 2016-2018 Confluent Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Refer to LICENSE for more information.
+
+using System.Net.Http.Headers;
+
+namespace Confluent.SchemaRegistry
+{
+    /// <summary>
+    ///     An interface defining Schema Registry HTTP client authentication schemes.
+    /// </summary>
+    public interface IHttpAuthenticator
```

Comment:
I think a more specific name here is better `IAuthenticationHeaderValueProvider`. Also update the comment.